creator-node/src/snapbackSM/snapbackSM.js

```diff
@@ -1004,6 +927,103 @@ class SnapbackSM {
     return clockValue
   }
 
+  /*
+   * For every node user, record sync requests to issue to secondaries if this node is primary
+   *    and record replica set updates to issue for any unhealthy replicas
+   *
+   * Purpose for the if/else case is that if the current node is a primary, issue reconfig or sync requests.
+   * Else, if the current node is a secondary, only issue reconfig requests.
+   *
+   * @notice this will issue sync to healthy secondary and update replica set away from unhealthy secondary
+   */
+  async aggregateReconfigAndPotentialSyncOps (nodeUsers, unhealthyPeers) {
+    const requiredUpdateReplicaSetOps = []
+    const potentialSyncRequests = []
+
+    for (const nodeUser of nodeUsers) {
+      const { primary, secondary1, secondary2, primarySpID, secondary1SpID, secondary2SpID } = nodeUser
+
+      let unhealthyReplicas = []
+
+      /**
+       * If this node is primary for user, check both secondaries for health
+       * Enqueue SyncRequests against healthy secondaries, and enqueue UpdateReplicaSetOps against unhealthy secondaries
+       */
+      let replicaSetNodesToObserve = [
+        { endpoint: secondary1, spId: secondary1SpID },
+        { endpoint: secondary2, spId: secondary2SpID }
+      ]
+
+      if (primary === this.endpoint) {
+        // filter out false-y values to account for incomplete replica sets
+        replicaSetNodesToObserve = replicaSetNodesToObserve.filter(entry => entry.endpoint && entry.spId)
+
+        /**
+         * If either secondary is in `unhealthyPeers` list, add it to `unhealthyReplicas` list
+         */
+        const userSecondarySyncMetrics = await this._computeUserSecondarySyncSuccessRates(nodeUser, [secondary1, secondary2])
+        for (const secondary of replicaSetNodesToObserve) {
+          const secUserSyncSuccessRate = userSecondarySyncMetrics[secondary.endpoint]['SuccessRate']
+          if (secUserSyncSuccessRate < this.MinimumSecondaryUserSyncSuccessPercent ||
+            unhealthyPeers.has(secondary.endpoint) ||
+            this.peerSetManager.endpointToSPIdMap[secondary.endpoint] !== secondary.spId) {
```

Comment:
i have done this for the above 2 cases in my current PR but for this new case (spID mismatch), we should do it as a standalone if check bc i want to issue a speicifc log statement that we can track

Reply:
given that an event of a mismatched spId due to reregistration is so low, do you think it's worth logging? I don't know if we'll catch many events even with a log

Reply:
i still think it'd be useful, but nw - i can handle this in my PR